lib/tasks/clean.rake

```diff
@@ -36,15 +36,33 @@ end
 
 desc 'Create measurements from series table and destroy the old ones'
 task :reset_measurements => :environment do
+  ActiveRecord::Base.connection.execute('UPDATE series SET measurement_id = NULL;')
   ActiveRecord::Base.connection.execute('DELETE FROM measurements;')
   ActiveRecord::Base.connection.execute(%Q|INSERT INTO
 measurements (prefix, data_portal_name, units_label, units_label_short, percent, `real`, created_at, updated_at)
-(SELECT LEFT(series.name, LOCATE('@', series.name) - 1) AS prefix, MAX(dataPortalName) AS data_portal_name,
+(SELECT
+  TRIM(TRAILING '&NS' FROM TRIM(TRAILING 'NS' FROM UPPER(LEFT(series.name, LOCATE('@', series.name) - 1)))) AS prefix,
+  MAX(dataPortalName) AS data_portal_name,
   MAX(unitsLabel) AS units_label, MAX(unitsLabelShort) AS units_label_short, MAX(percent) AS percent,
   MAX(series.real) AS `real`, NOW(), NOW()
-FROM series GROUP BY LEFT(series.name, locate('@', series.name) - 1));|)
+FROM series GROUP BY
+  TRIM(TRAILING '&NS' FROM TRIM(TRAILING 'NS' FROM UPPER(LEFT(series.name, LOCATE('@', series.name) - 1))))
```

Comment:
MySQL will allow you to GROUP BY 1 here, so you don't have to repeat the whole expression. Not sure whether you avoid that as a matter of principle, but just sayin :=)